### ISSUE TITLE
Issue 9/84

### ISSUE THREAD

**Issue by mist64** [mist64/64er-magazin.de#74]
https://www.64er-magazin.de/test/issue/8409/8409/

- [x] scan magazine (2400 dpi PNG files)
- [x] PDF
	- [x] `widths.txt` (cropping info)
	- [x] `title.png` (retouched 150 dpi title page)
	- [x] `64er_19xx_xx.pdf` (OCR'ed PDF)
- [x] Web
	- [x] `all.md` (all OCR'ed text)
	- [x] `tables.html` (all OCR'ed tables)
	- [x] convert .md to .html
		- [x] `<p class="intro">`
		- [x] `<ul>`/`<ol>`
		- [x] add `<address class="author">` around author
	- [x] split into individual .html files
	- [x] fill tables from `tables.html`
	- [x] Fehlerteufelchen
		- [x]  add `<aside>`s
		- [x] link using `<a href="#fehlerteufelchen" class="fehlerteufelchen_link">`
	- [x] format articles
		- [x] `<aside>` tag for author bio, `<img class="inline">` for author photo
		- [x] `<p class="literatur">` for fine print at end of article
	- [x] special articles
		- [x] Editorial: `<img class="inline">` for editor's photo
		- [x] Leserforum: markup and image title
		- [x] Bücher: image title
		- [x] Impressum: base on previous issue; apply diff
	- [x] metadata
		- [x] copy title into `title` tag in header
		- [x] copy author into `author` tag in header
		- [x] fill `64er.id` tag (lowercase super short form of title)
		- [x] fill `64er.toc_category` tag from table of contents
		- [x] fill `64er.pages` tag (list/ranges of page numbers)
		- [x] fill `64er.head1`/`64er.head2` tags from magazine page headers
		- [x] fill `64er.index_category` and `64er.index_title` from Jahresinhaltsverzeichnis CSV
	- [x] images
		- [x] extract cropped images, save as PNG (150 color, 600 b/w)
		- [x] convert b/w images to duotone
		- [x] insert image links and captions
	- [x] listings
		- [x] `<figure>` tags
		- [x] add petcat format listing 
		- [x] link listings
	- [x] `toc.txt` (order of `64er.toc_category` keys in table of contents)
	- [x] `pubdate.txt` (publication date)

**Comment by thierer** [comment]
I'll add the `pages` and `head1`/`head2` tags.

Btw, where did you get the CSV with the toc entries from?

**Comment by mist64** [comment]
> Btw, where did you get the CSV with the toc entries from?

I created the CVS from the Jahresinhaltsverzeichnis in 4/85 and put it in the root of the repo. I also wrote a script (it's in `tools`) that generates meta tags from the CVS. I checked in a file `index_headers.txt` into every branch with the result from the script, which looks like this:

```
8503, S.8
    <meta name="64er.index_title" content="Heiße Messe in der Wüste: CES (PC 128, PC 10, Commodore LCD)">
    <meta name="64er.index_category" content="Aktuell|Computer">
8503, S.14
    <meta name="64er.index_title" content="Neues 1451-Laufwerk">
    <meta name="64er.index_category" content="Aktuell|Floppy">
8503, S.15
    <meta name="64er.index_title" content="Internationaler Chaos Communiction Congress">
    <meta name="64er.index_category" content="Aktuell|DFÜ">
[...]
```
So it's easy to copy and paste the two lines each into each file.

**Comment by thierer** [comment]
I understood most of that, I just wondered where it came from. Did it already exist somewhere in a machine-readable form? Probably not, as you write you "created" it. So did you type it in or were you able to OCR and convert it?

**Comment by mist64** [comment]
I OCRed it, using FineReader on Windows. Tables work quite well, if you do some manual adjustments. I loaded non-color-corrected 600 dpi scans, marked the tables, let it automatically analyze the table structure, fixed up the table structure (i.e. moved a few line and column separators), then let it recognize the text, and saved it as HTML. I loaded the HTML into Apple Numbers, corrected all OCR issues I could find and fixed up the layout.
